src/vs/workbench/contrib/terminal/browser/terminalProfileQuickpick.ts

```diff
@@ -41,9 +41,50 @@ export class TerminalProfileQuickpick {
 			return;
 		}
 		if (type === 'setDefault') {
+			// Check if workspace and user default profiles differ
+			const configInspect = this._configurationService.inspect(defaultProfileKey);
+			const userDefault = configInspect.userValue;
+			const workspaceDefault = configInspect.workspaceValue;
```

Comment:
Testing this it doesn't work as I expected.

Selecting a workspace profile will carry the workspace profile over to user settings, then set it. Instead we want to only set it as the default in the workspace settings in that case.